sdk/servicebus/azure_messaging_servicebus/src/error.rs

```diff
@@ -53,16 +53,16 @@ impl fmt::Display for ErrorKind {
 }
 
 /// A Service Bus specific error.
-#[derive(SafeDebug, Clone, PartialEq, Eq)]
+#[derive(SafeDebug)]
 pub struct ServiceBusError {
     kind: ErrorKind,
     message: String,
-    source: Option<Box<ServiceBusError>>,
+    source: Option<Box<dyn std::error::Error + Send + Sync + 'static>>,
```

Comment:
Why add `Send`, `Sync`, and `'static` constraints? That makes it pretty difficult to send any error since not all implement those traits.